src/pluginHooks.js

```diff
@@ -222,6 +222,16 @@ const REGISTERED_HOOKS = [
    */
   'afterCreateRow',
 
+  /**
+   * Fired before splicing the data during row creation process.
+   *
+   * @event Hooks#beforeDataSplice
+   * @param {number} index Physical index of the element to add/remove.
+   * @param {number} amount Number of rows to add/remove.
+   * @param {object[]} elements Array of row elements to be added.
```

Comment:
Related to the previous question. Maybe we shouldn't inform a user that this hook exists? I know that it's public. However, we have some not mentioned hooks like i.e. `modifyRemovedAmount`, `afterIsMultipleSelection`, `postAfterValidate`.